kwrooijen/gungnir
# WIP: Migration Proof of Concept

Don't merge, it's just a proof of concept to test out ideas.

It tried to recreate the conditions laid out in #2 

The first migration wasn't possible, I had an issue with loading ossp-uuid, something about versions being different, but I tested and the [:raw] tag works as expected.

The second migration was slightly adapted for 2 reasons: 
1) postgres doesn't like making a user table (i mean you can make it but it's going to be annoying because you always have to have it in quotations)

2) the :table.add/column key was changed to just table/column, so it more closely resembles the protocol used in [honeysql-postgres](https://github.com/nilenso/honeysql-postgres#create-table)

I added a small folder with the migrations in a mostly random location

to test the existing migrations out:
- make sure gungnir.database/*datasource* has been defined,
- run `(gungnir.database/make-datasource! with-valid-arguments)`
- run `(gungnir.migration/migrate-all)`

to test creating a migration use `(gungnir.migration/create-migration name content)`
